### PR TITLE
fix(go): add auth

### PR DESCRIPTION
**Pull request by divyenduz**
Add auth support for Prisma service protected with a secret.

### CODE REVIEW

File: client/src/codegen/go-client.ts [hunk 1]
```diff
@@ -705,16 +723,23 @@ type Client struct {
 }
 
 type Options struct {
-	Endpoint string
```
Comment (dominikh) [inline client/src/codegen/go-client.ts]: Inconsistent indentation, mixing tabs and spaces. Before the change, all the code in the `static` string was indented with tabs.

File: client/src/codegen/go-client.ts [hunk 2]
```diff
@@ -705,16 +723,23 @@ type Client struct {
...
+  Secret   string
 }
 
 func New(options *Options, opts ...graphql.ClientOption) *Client {
-	endpoint := DefaultEndpoint
+  endpoint := DefaultEndpoint
+  secret   := Secret
 	if options != nil {
-		endpoint = options.Endpoint
+    if options.Endpoint != "" {
```
Comment (dominikh) [inline client/src/codegen/go-client.ts]: Has what I discussed with @schickling changed? We had agreed that if the user provided a non-nil `options`, that all values would have to be provided, and not have defaults.

File: client/src/codegen/go-client.ts [hunk 3]
```diff
@@ -650,6 +650,24 @@ export class GoGenerator extends Generator {
     }
   }
 
+  printSecret(options: RenderOptions): string | null {
+    if (!options.secret) {
+      return null
```
Comment (dominikh) [inline client/src/codegen/go-client.ts]: Wouldn't this branch result in us printing `var Secret = null`? That wouldn't be correct.

File: client/src/codegen/go-client.ts (hunk 2 above)
Comment (divyenduz) [reply client/src/codegen/go-client.ts]: Indeed, you are correct, I will make amends.

**Pull request merged**